ompi/datatype/ompi_datatype_args.c

```diff
@@ -78,33 +78,33 @@ typedef struct __dt_args {
  * setting the 64 bits aligned one we will not have any trouble. Problem
  * originally reported on SPARC 64.
  */
-#define ALLOC_ARGS(PDATA, IC, AC, DC)                                   \
-    do {                                                                \
-        int length = sizeof(ompi_datatype_args_t) + (IC) * sizeof(int) + \
-            (AC) * sizeof(ptrdiff_t) + (DC) * sizeof(MPI_Datatype); \
-        char* buf = (char*)malloc( length );                            \
-        ompi_datatype_args_t* pArgs = (ompi_datatype_args_t*)buf;       \
-        pArgs->ci = (IC);                                               \
-        pArgs->ca = (AC);                                               \
-        pArgs->cd = (DC);                                               \
-        buf += sizeof(ompi_datatype_args_t);                            \
-        if( pArgs->ca == 0 ) pArgs->a = NULL;                           \
-        else {                                                          \
-            pArgs->a = (ptrdiff_t*)buf;                         \
-            buf += pArgs->ca * sizeof(ptrdiff_t);               \
-        }                                                               \
-        if( pArgs->cd == 0 ) pArgs->d = NULL;                           \
-        else {                                                          \
-            pArgs->d = (ompi_datatype_t**)buf;                          \
-            buf += pArgs->cd * sizeof(MPI_Datatype);                    \
-        }                                                               \
-        if( pArgs->ci == 0 ) pArgs->i = NULL;                           \
-        else pArgs->i = (int*)buf;                                      \
-        pArgs->ref_count = 1;                                           \
-        pArgs->total_pack_size = (4 + (IC) + (DC)) * sizeof(int) +      \
-            (AC) * sizeof(ptrdiff_t);                                   \
-        (PDATA)->args = (void*)pArgs;                                   \
-        (PDATA)->packed_description = 0;                                \
+#define ALLOC_ARGS(PDATA, IC, AC, DC)                                         \
+    do {                                                                      \
+        int length = sizeof(ompi_datatype_args_t) + (IC) * sizeof(int) +      \
+            (AC) * sizeof(ptrdiff_t) + (DC) * sizeof(MPI_Datatype);           \
+        char* buf = (char*)malloc( length );                                  \
+        ompi_datatype_args_t* alloc_args_tmp  = (ompi_datatype_args_t*)buf;   \
+        alloc_args_tmp = (IC);                                                \
```

Comment:
This is different than the original code.

Reply:
Thanks - I think I fixed it with the latest push.